packages/edit-post/src/components/header/header-toolbar/index.js

```diff
@@ -65,23 +68,50 @@ function HeaderToolbar( { onToggleInserter, isInserterOpen } ) {
 			className="edit-post-header-toolbar"
 			aria-label={ toolbarAriaLabel }
 		>
-			<Button
-				className="edit-post-header-toolbar__inserter-toggle"
-				isPrimary
-				isPressed={ isInserterOpen }
-				onClick={ onToggleInserter }
-				disabled={ ! isInserterEnabled }
-				icon={ plus }
-				label={ _x(
-					'Add block',
-					'Generic label for block inserter button'
+			<ToolbarItem>
+				{ ( itemProps ) => (
+					<Button
+						{ ...itemProps }
+						className="edit-post-header-toolbar__inserter-toggle"
+						isPrimary
+						isPressed={ isInserterOpen }
+						onClick={ onToggleInserter }
+						disabled={ ! isInserterEnabled }
+						icon={ plus }
+						label={ _x(
+							'Add block',
+							'Generic label for block inserter button'
+						) }
+					/>
+				) }
+			</ToolbarItem>
+			{ isLargeViewport && (
+				<ToolbarItem>
+					{ ( itemProps ) => <ToolSelector { ...itemProps } /> }
+				</ToolbarItem>
+			) }
+			<ToolbarItem>
+				{ ( itemProps ) => <EditorHistoryUndo { ...itemProps } /> }
+			</ToolbarItem>
+			<ToolbarItem>
+				{ ( itemProps ) => <EditorHistoryRedo { ...itemProps } /> }
+			</ToolbarItem>
+			<ToolbarItem>
+				{ ( itemProps ) => (
+					<TableOfContents
+						hasOutlineItemsDisabled={ isTextModeEnabled }
+						{ ...itemProps }
+					/>
+				) }
+			</ToolbarItem>
+			<ToolbarItem>
+				{ ( itemProps ) => (
+					<BlockNavigationDropdown
+						isDisabled={ isTextModeEnabled }
```

Comment:
Shouldn't the order be reversed to ensure that props from `itemProps` never override `isDisabled`? I guess it's mostly to stay on the safe side knowing that those APIs evolve over time.

Also the question about using `as` applies here.